tests/rules/test_document_end.py

```diff
@@ -71,3 +71,22 @@ def test_multiple_documents(self):
                    '---\n'
                    'third: document\n'
                    '...\n', conf, problem=(6, 1))
+
+    def test_directives(self):
+        conf = 'document-end: {present: true}'
+        self.check('%YAML 1.2\n'
+                   '---\n'
+                   'document: end\n'
+                   '...\n', conf)
+        self.check('%YAML 1.2\n'
+                   '%TAG ! tag:clarkevans.com,2002:\n'
+                   '---\n'
+                   'document: end\n'
+                   '...\n', conf)
+        self.check('---\n'
+                   'first: document\n'
+                   '...\n'
+                   '%YAML 1.2\n'
+                   '---\n'
+                   'second: document\n'
+                   '...\n', conf)
```

Comment:
We could also check that yamlllint complains if the `...` before `%YAML 1.2` are missing. Looking at [the YAML spec](https://yaml.org/spec/1.2.2/), it seems forbidden to do so:

```yaml
---
first: document
%YAML 1.2
---
second: document
...
```

But maybe this isn't really related to this pull request.

Reply:
Not directly related, but still belongs to `test_directives`.